src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -204,9 +204,6 @@ public TransactionSubstate Run<TTracingActions>(EvmState state, IWorldState worl
                             // TODO: when direct / calls are treated same we should not need such differentiation
                             goto Failure;
                         }
-
-                        // TODO: testing it as it seems the way to pass zkSNARKs tests
-                        currentState.GasAvailable = 0;
```

Comment:
that looks like a consensus change
@benaadams

Reply:
zk Ethereum tests do indeed seem to fail with this removed as per comment